src/optimizer.cpp

```diff
@@ -230,18 +247,25 @@ void Optimizer::propagateStateVelocitiesFromInitials(
 {
   using namespace xt::placeholders;  // NOLINT
 
-  for (size_t i = 0; i < settings_.time_steps - 1; i++) {
-    auto curr_state = xt::view(state.data, xt::all(), i, xt::all());
-    auto next_velocities =
-      xt::view(
-      state.data, xt::all(), i + 1,
-      xt::range(state.idx.vbegin(), state.idx.vend()));
+  if (motion_model_->isNaive()) {
```

Comment:
I think it would still be better to do this velocity setting on controls in the `predict` function. I don't see why we need to re-introduce it here

Reply:
Essentially this change makes a new feature `naive` which there is no alternative of for evaluation in the optimizer, nor is there any implementation that uses in the motion model (and both throw exceptions if you try). Lets simplify it back to how it was beforehand.

Reply:
done